kcidb/monitor/subscriptions/linux_stable_rc.py

```diff
@@ -0,0 +1,35 @@
+"""Linux stable-rc subscription"""
+from datetime import (timezone, datetime, timedelta)
+
+from kcidb.monitor.output import NotificationMessage as Message
+
+
+def match_revision(revision):
+    """Match revisions of stable-rc tree"""
+    # Repo we're interested in
+    repo_url = "https://git.kernel.org/pub/scm/linux/kernel/git/stable/\
+linux-stable-rc.git"
+
+    # If the revision is not from stable-rc repo,
+    # or there are no finished builds
+
+    if repo_url not in revision.repo_branch_checkouts:
+        return ()
+
+    if revision.builds_valid is None:
+        return ()
+
+    # We need to send the notification one hour after the last build came in.
+    # However, we currently have no way of distinguishing build updates
+    # from updates to build's tests, and we can't send a notification
+    # about a revision from a build update yet. So we're sending the
+    # notification one hour after any revision update.
```

Comment:
Does this comment really apply to what you wanted to do here, as is? If not, please describe what you actually want and/or what you're doing.

Reply:
Um, somehow I thought I already responded to this comment, but can't find it. Sorry, if you get it twice.

The comment implies that you somehow *wanted* to have notifications *only* after build updates. Was that really the case? If not, please remove that from the comment.

Reply:
I think we should send the report once all builds and tests are received for the particular checkout.
In the case of maestro, we set checkout node's `valid` flag to `True`  once all the build and test executions have been completed. 
But that might not be true for other CI systems.
I think we should wait longer than 1 hr after a build is updated.
Do we currently have any frequency set for sending `stable-rc` report that you generate manually? @crazoes

Reply:
Yeah, we cannot expect anybody else to do the same thing with checkouts. Also such use is really a hack, as `valid` means that you've managed to apply the referenced patch. Nothing else. And you don't have patches, so it should always be valid.

Whatever logic you find out works for you, please put the corresponding comment over it, don't leave the original. It pertains to the specific plan and workaround we had with Mark.

Reply:
@JenySadadia we send the report once Greg sends out an email notification of the release. So for stable-rc specifically we need to parse the stable-rc tree and whenever there is a new changes pushed, we need to generate a report and send it to me initially. I will then forward the email after verification.

I did work on a script to parse the stable-rc tree long time back :- 
https://gitlab.collabora.com/shreeya/kernelci/-/blob/main/email_checker.py

You can integrate this script to generate a report accordingly.

Reply:
Polling for an email in the inbox looks a bit tricky here.
Can we integrate this kind of script in KCIDB? @spbnick

Reply:
I assume we want to send our results *not before* Greg's announcement, and preferably in response to it, right?

KCIDB subscriptions are done for database changes only. There's no object for a " particular message posted on a particular maillist" (or perhaps a "release announcement") in KCIDB database, and therefore no event on its appearance/update like the rest of the data. And I don't think we should add it for Greg only, even though his work is very important.

I think our plan for now could be "generate a notification when done", and let a human decide when to send it, as at the moment we require a human for review anyway. Then, when we're in a better place, and there's not so much work to do everywhere, we could think of making it better.

Reply:
We still need to fix this comment before merging.

Reply:
Maestro usually allows a checkout to be completed with all builds and tests within 3 hrs. But of course, that time would differ per CI system basis.
So, I guess we can set the `due` time accordingly.
Bdw list of revisions is managed internally by KCIDB. So when is `match_revision` event triggered? On every checkout update or build update?

Reply:
Gentle ping ^ @spbnick

Reply:
Sorry, missed this question. `match_revision` is triggered on any new/updated checkout/build/test arriving for the particular revision.

Reply:
Okay. So, it means it will create a new report on every build or test submission.
In that case, we will have hundreds of reports (because of test submissions) for the same checkout.
And I don't think we want that in this case.
We can restrict it to listening to only checkout events and `match_checkout` won't work I guess as it won't contain any build or test data.
What do you suggest?

Reply:
The function is triggered on every update, yes. However it creates/updates a notification with the ID of the object it triggered on, the revision ID, plus the subscription name, and a notification with a particular ID is only sent *once* by the system. So on every update it will create/update a notification with the same ID, scheduled to be sent in an hour (or however long you want). This way the sending time is pushed forward by updates until the "last" update.

Reply:
I.e. the function *creates* or *updates* a notification, it doesn't send it.

Reply:
Thank you for the clarification.
I have set `3 hrs` delay for sending a notification based on some observations. We can tweak it once we have some real-time notifications.
Updated the comment accordingly.